src/tree.rs

```diff
@@ -436,6 +438,147 @@ impl<T> Tree<T> {
     }
 
     ///
+    /// Sorts the children of one node, in-place, using compare to compare the nodes
+    ///
+    /// This sort is stable and O(n log n) worst-case but allocates approximately 2 * n where n is the length of children
+    ///
+    /// Returns an empty `Result` containing a `NodeIdError` if one occurred.
+    ///
+    /// ```
+    /// use id_tree::Tree;
+    /// use id_tree::Node;
+    ///
+    /// let root_node = Node::new(100);
+    /// let first_child = Node::new(1);
+    /// let second_child = Node::new(2);
+    /// let third_child = Node::new(0);
+    ///
+    /// let mut tree: Tree<i32> = Tree::new();
+    /// let root_id = tree.set_root(root_node);
+    ///
+    /// tree.insert_with_parent(first_child, &root_id).unwrap();
+    /// tree.insert_with_parent(second_child, &root_id).unwrap();
+    /// tree.insert_with_parent(third_child, &root_id).unwrap();
+    ///
+    /// tree.sort_children_by(&root_id, |a, b| a.data().cmp(b.data()));
+    ///
+    /// # for (i, id) in tree.get(&root_id).unwrap().children().iter().enumerate() {
+    /// #   assert_eq!(*tree.get(&id).unwrap().data(), i as i32);
+    /// # }
+    /// ```
+    ///
+    pub fn sort_children_by<F>(&mut self, node_id: &NodeId, mut compare: F) -> Result<(), NodeIdError>
+        where F: FnMut(&Node<T>, &Node<T>) -> Ordering
+    {
```

Comment:
Nope, I agree with you, it looks quite strange with a `where` clause.  You have it exactly how I would have done it.

Also, could you point me to the docs for rustfmt?  I followed your link earlier (and did some searching as well), but couldn't find anything other than their README.md.  I'm specifically looking for what can be configured in the rustfml.toml file.